### PR TITLE
[wallet-ext] Add support for fetching pending delegations

### PR DESCRIPTION
**Pull request by Jordan-Mysten**
This updates the wallet staking flow to display pending delegations, fetching them through the system state object. This way of fetching them is pretty rough / hacky, but hopefully it's something that can be improved down the road. Currently, I'm displaying these as "in warm-up", although I imagine this could change once warmup and cooldown are actually implemented in contract.

### CODE REVIEW

File: apps/wallet/src/ui/app/staking/usePendingDelegation.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+                    validator.fields.delegation_staking_pool.fields
+                        .pending_delegations;
+
+                if (!Array.isArray(pendingDelegations)) return null;
+
+                const filteredDelegations = pendingDelegations.filter(
+                    (delegation) => delegation.fields.delegator === address
+                );
+
+                if (!filteredDelegations) return null;
```
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/staking/usePendingDelegation.tsx:90]: Would `filteredDelegations` always be an array?

File: apps/wallet/src/ui/app/staking/usePendingDelegation.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                    staked: filteredDelegations.reduce(
+                        (acc, delegation) =>
+                            acc + BigInt(delegation.fields.sui_amount),
+                        0n
+                    ),
+                } as PendingDelegation;
+            })
+            .filter(notEmpty);
+
+        return validators;
```
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/staking/usePendingDelegation.tsx:109]: `validators` is a bit confusing to me - this is pending delegations per validator right?

**Review by pchrysochoidis — APPROVED** [review]

File: apps/wallet/src/ui/app/staking/usePendingDelegation.tsx (hunk 1 above)
Comment (Jordan-Mysten) [reply apps/wallet/src/ui/app/staking/usePendingDelegation.tsx:90]: Good catch, will fix

File: apps/wallet/src/ui/app/staking/usePendingDelegation.tsx (hunk 2 above)
Comment (Jordan-Mysten) [reply apps/wallet/src/ui/app/staking/usePendingDelegation.tsx]: Good point, updated

**Pull request merged**